File: docs/history.py
from db import db

def log_user_summary(user_id, video_id, summary, transcript, language):
    doc_ref = db.collection("summaries").document(f"{user_id}_{video_id}")
    doc_ref.set({
        "user_id": user_id,
        "video_id": video_id,
        "summary": summary,
        "transcript": transcript,
        "language": language
    })
    return True

File: docs/history.js


const db = require('./db.js');

function logUserSummary(userId, videoId, summary, transcript, language) {
  const docRef = db.collection('summaries').doc(`${userId}_${videoId}`);

  return docRef.set({
    user_id: userId,
    video_id: videoId,
    summary: summary,
    transcript: transcript,
    language: language
  })
  .then(() => true)
  .catch((error) => {
    console.error('Error writing document:', error);
    return false;
  });
}

module.exports = { logUserSummary };

